rust/src/mask/masking.rs

```diff
@@ -0,0 +1,529 @@
+use anyhow::{anyhow, Context};
+use rand::{RngCore, SeedableRng};
+use std::{
+    convert::TryInto,
+    iter::{self, Iterator},
+    ops::Range,
+};
+
+use num::{
+    bigint::{BigInt, BigUint, ToBigInt},
+    clamp,
+    rational::Ratio,
+    traits::Zero,
+};
+use rand_chacha::ChaCha20Rng;
+
+use crate::{
+    mask::{config::MASK_CONFIG_BUFFER_LEN, MaskConfig, MaskSeed, Model},
+    message::{utils::range, DecodeError, FromBytes, ToBytes},
+};
+
+use thiserror::Error;
+
+#[derive(Debug, Hash, PartialEq, Eq, Clone)]
+pub struct MaskObject {
+    data: Vec<BigUint>,
+    config: MaskConfig,
+}
+
+impl MaskObject {
+    pub fn new(config: MaskConfig, data: Vec<BigUint>) -> Self {
+        Self { data, config }
+    }
+
+    pub fn new_empty(config: MaskConfig) -> Self {
+        Self {
+            data: vec![],
```

Comment:
`data: Vec::new()` results in the same, but would avoid the macro.

Reply:
Ok I'll change it.